src/dict.c

```diff
@@ -532,6 +532,15 @@ int dictAdd(dict *d, void *key, void *val) {
     return DICT_OK;
 }
 
+/* Add an element with double value to the target hash table */
+int dictAddDoubleVal(dict *d, void *key, double val) {
+    dictEntry *entry = dictAddRaw(d, key, NULL);
+
+    if (!entry) return DICT_ERR;
+    if (!d->type->no_value) dictSetDoubleVal(entry, val);
+    return DICT_OK;
+}
+
```

Comment:
You don't need to introduce this new method and expose it. Rather you could do it in a pair of operation, create a dict entry and set the value.

```
dictAddRaw(...);
dictSetDoubleVal(...);

```

Reply:
Thanks for your advice, the `dictSetDoubleVal` function has been removed.